--- ../vendor/nikic/php-parser/grammar/parser.template	2018-10-10 18:24:14.000000000 +0900
+++ grammar/parser.template	2019-03-20 19:25:32.000000000 +0900
@@ -5,14 +5,17 @@
 #semval(%n) $stackPos-(%l-%n)
 #semval(%n,%t) $stackPos-(%l-%n)
 
-namespace PhpParser\Parser;
+namespace Phinder\Parser\PatternParser;
 
-use PhpParser\Error;
-use PhpParser\Node;
-use PhpParser\Node\Expr;
-use PhpParser\Node\Name;
-use PhpParser\Node\Scalar;
-use PhpParser\Node\Stmt;
+use \PhpParser\Error;
+use \PhpParser\Node;
+use \PhpParser\Node\Expr;
+use \PhpParser\Node\Name;
+use \PhpParser\Node\Scalar;
+use \PhpParser\Node\Stmt;
+use Phinder\Wildcard;
+use Phinder\WildcardN;
+use Phinder\ArrayItem;
 #include;
 
 /* This is an automatically GENERATED file, which should not be manually edited.
