--- ../vendor/nikic/php-parser/grammar/php7.y	2018-09-18 16:03:24.000000000 +0900
+++ grammar/php7.y	2018-10-24 18:33:41.000000000 +0900
@@ -6,7 +6,7 @@
 %%
 
 start:
-    top_statement_list                                      { $$ = $this->handleNamespaces($1); }
+      expr                                               { $$ = $1; }
 ;
 
 top_statement_list_ex:
@@ -234,7 +234,6 @@
     | T_STATIC static_var_list semi                         { $$ = Stmt\Static_[$2]; }
     | T_ECHO expr_list semi                                 { $$ = Stmt\Echo_[$2]; }
     | T_INLINE_HTML                                         { $$ = Stmt\InlineHTML[$1]; }
-    | expr semi                                             { $$ = Stmt\Expression[$1]; }
     | T_UNSET '(' variables_list ')' semi                   { $$ = Stmt\Unset_[$3]; }
     | T_FOREACH '(' expr T_AS foreach_variable ')' foreach_statement
           { $$ = Stmt\Foreach_[$3, $5[0], ['keyVar' => null, 'byRef' => $5[1], 'stmts' => $7]]; }
@@ -492,6 +491,7 @@
 argument:
       expr                                                  { $$ = Node\Arg[$1, false, false]; }
     | '&' variable                                          { $$ = Node\Arg[$2, true, false]; }
+    | T_ELLIPSIS                                            { $$ = Node\Arg[WildcardN[], false, false]; }
     | T_ELLIPSIS expr                                       { $$ = Node\Arg[$2, false, true]; }
 ;
 
@@ -702,6 +702,7 @@
     | T_INCLUDE expr                                        { $$ = Expr\Include_[$2, Expr\Include_::TYPE_INCLUDE]; }
     | T_INCLUDE_ONCE expr                                   { $$ = Expr\Include_[$2, Expr\Include_::TYPE_INCLUDE_ONCE]; }
     | T_EVAL '(' expr ')'                                   { $$ = Expr\Eval_[$3]; }
+    | T_EVAL                                                { $$ = Expr\Eval_[Wildcard[]]; }
     | T_REQUIRE expr                                        { $$ = Expr\Include_[$2, Expr\Include_::TYPE_REQUIRE]; }
     | T_REQUIRE_ONCE expr                                   { $$ = Expr\Include_[$2, Expr\Include_::TYPE_REQUIRE_ONCE]; }
     | T_INT_CAST expr                                       { $$ = Expr\Cast\Int_    [$2]; }
@@ -891,6 +892,7 @@
       callable_variable                                     { $$ = $1; }
     | static_member                                         { $$ = $1; }
     | dereferencable T_OBJECT_OPERATOR property_name        { $$ = Expr\PropertyFetch[$1, $3]; }
+    | '?'                                                   { $$ = Wildcard[]; }
 ;
 
 simple_variable:
@@ -971,6 +973,7 @@
 array_pair:
       expr T_DOUBLE_ARROW expr                              { $$ = Expr\ArrayItem[$3, $1,   false]; }
     | expr                                                  { $$ = Expr\ArrayItem[$1, null, false]; }
+    | T_ELLIPSIS                                            { $$ = Expr\ArrayItem[WildcardN[], null, false]; }
     | expr T_DOUBLE_ARROW '&' variable                      { $$ = Expr\ArrayItem[$4, $1,   true]; }
     | '&' variable                                          { $$ = Expr\ArrayItem[$2, null, true]; }
     | /* empty */                                           { $$ = null; }
