--- ../vendor/nikic/php-parser/grammar/rebuildParsers.php	2018-10-10 18:24:14.000000000 +0900
+++ grammar/rebuildParsers.php	2019-03-20 19:24:50.000000000 +0900
@@ -1,8 +1,9 @@
 <?php
+error_reporting(E_ALL ^ E_WARNING);
 
 $grammarFileToName = [
-    __DIR__ . '/php5.y' => 'Php5',
-    __DIR__ . '/php7.y' => 'Php7',
+    __DIR__ . '/php5.y' => 'Pattern5',
+    __DIR__ . '/php7.y' => 'Pattern7',
 ];
 
 $tokensFile     = __DIR__ . '/tokens.y';
@@ -10,7 +11,7 @@
 $skeletonFile   = __DIR__ . '/parser.template';
 $tmpGrammarFile = __DIR__ . '/tmp_parser.phpy';
 $tmpResultFile  = __DIR__ . '/tmp_parser.php';
-$resultDir = __DIR__ . '/../lib/PhpParser/Parser';
+$resultDir = __DIR__ . '/../lib/Parser/PatternParser';
 $tokensResultsFile = $resultDir . '/Tokens.php';
 
 // check for kmyacc.exe binary in this directory, otherwise fall back to global name
